### PR TITLE
Adopt meta DEx reserved balances on RPC layer

### PR DESCRIPTION
**Pull request by dexX7**
With PR #223 all regtests pass, for the first time! ;)

```
Test summary:

ExodusPurchaseTest ... successful
PropertyCreationTest ... successful
P2SHTest ... successful
MetaDexPlanTest ... successful
MetaDexCancelAtPriceTest ... successful
MetaDexCancelPairAndLookupTest ... successful
MetaDexCancelEverythingInSameEcosystemTest ... successful
MetaDexCancelEverythingIgnorePropertyTest ... successful
MetaDexCancelEverythingScopeTest ... successful
DexCrossEcosystemSideEffectsTest ... successful
TraditionalDexTest ... successful

11 of 11 tests successful
```

https://gist.githubusercontent.com/dexX7/061c73f7d4eec0a2e767/raw/

### CODE REVIEW

**Comment by dexX7** [comment]
Minimized and seperated from #236.

File: src/mastercore_rpc.cpp [hunk 1]
```diff
@@ -84,6 +84,20 @@ void MetaDexObjectsToJSON(std::vector<CMPMetaDEx>& vMetaDexObjs, Array& response
     }
 }
 
+void BalanceToJSON(const std::string& address, uint32_t property, Object& balance_obj)
+{
+    // balance without PENDING amount
+    int64_t nAvailable = getUserAvailableMPbalance(address, property);
+
```
Comment (zathras-crypto) [inline src/mastercore_rpc.cpp]: getUserAvailableMPBalance returns balance _with_ pending amount deductions FYI

**Comment by zathras-crypto** [comment]
Again looks like nice simplification removing some of the code duplication.  Think the 'without pending' is just a bad comment, actual code does include pending.  LGTM

File: src/mastercore_rpc.cpp (hunk 1 above)
Comment (dexX7) [inline src/mastercore_rpc.cpp]: Hehe you got me. Isn't this a question of how you read it?
`PENDING` is negative and `AVAILABLE` = `BALANCE` - `PENDING`.

Comment (zathras-crypto) [inline src/mastercore_rpc.cpp]: Haha I see; "balance without pending" = "balance minus pending" != "balance without including pending"

Comment (dexX7) [inline src/mastercore_rpc.cpp]: Strictly: `AVAILABLE` = `BALANCE` + `PENDING` where `PENDING` is actually negative.

A comment like "BALANCE + PENDING" would imply it reports unconfirmed, incoming amounts, not subtracting outgoing ones. Hm..

**Comment by dexX7** [comment]
Updated to:

``` c++
// confirmed balance minus unconfirmed, spent amounts 
```

... but any alternative is welcomed. :)

**Comment by dexX7** [comment]
Side note: this isn't only cosmetic, but also includes `METADEX_RESERVE` which was not the case earlier, resulting in incorrect balances.

**Comment by zathras-crypto** [comment]
Yep, assumed this was adding in the extra reserved tally that Michael split off to avoid conflict with DEx :)

Updated wording is fine mate...

**Pull request merged**